### PR TITLE
change IS_DMC from bit flag to bool

### PR DESCRIPTION
**Pull request by WalterBright**
Having it as a bit flag is just pointless complexity. It won't save a meaningful amount of memory. The other bit flags need to be converted to bool as well.

### CODE REVIEW

**Comment by RazvanN7** [comment]
> It won't save a meaningful amount of memory

What do you mean? Right now it doesn't use any memory, whereas if you transform all of its fields into bools, you will end up with 5 extra bytes per instance. It does seem like there are not many instances of `VisualCPPMangler` created during a compilation cycle, however, I don't really see the benefit of this change, I don't see the usage of an `enum` as pointless complexity.

File: compiler/src/dmd/cppmanglewin.d [hunk 1]
```diff
@@ -109,22 +109,21 @@ private final class VisualCPPMangler : Visitor
...
         ESCAPE = 0x10,
     }
 
     alias IS_NOT_TOP_TYPE = Flags.IS_NOT_TOP_TYPE;
     alias MANGLE_RETURN_TYPE = Flags.MANGLE_RETURN_TYPE;
     alias IGNORE_CONST = Flags.IGNORE_CONST;
-    alias IS_DMC = Flags.IS_DMC;
     alias ESCAPE = Flags.ESCAPE;
 
+    bool IS_DMC;        // Digital Mars C++ mangling
```
Comment (dkorpel) [inline compiler/src/dmd/cppmanglewin.d:120]:
```suggestion
    bool isDmc;        // Digital Mars C++ mangling
```
Variable field name should be camelCase

**Review by dkorpel — CHANGES_REQUESTED** [review]

**Comment by dkorpel** [comment]
> I don't see the usage of an enum as pointless complexity.

It's not the enum, it's packing multiple booleans into a single field, requiring the use of bitwise operators to read/write them instead of simple field access.

**Comment by WalterBright** [comment]
Merged into https://github.com/dlang/dmd/pull/15058

**Pull request closed without merging**